common/src/java/org/apache/hadoop/hive/common/type/TimestampTZUtil.java

```diff
@@ -145,13 +149,33 @@ public static ZoneId parseTimeZone(String timeZoneStr) {
     }
   }
 
+  public static Timestamp convertTimestampToZone(Timestamp ts, ZoneId fromZone, ZoneId toZone) {
+    return convertTimestampToZone(ts, fromZone, toZone, false);
+  }
+
   /**
    * Timestamps are technically time zone agnostic, and this method sort of cheats its logic.
    * Timestamps are supposed to represent nanos since [UTC epoch]. Here,
    * the input timestamp represents nanoseconds since [epoch at fromZone], and
    * we return a Timestamp representing nanoseconds since [epoch at toZone].
    */
-  public static Timestamp convertTimestampToZone(Timestamp ts, ZoneId fromZone, ZoneId toZone) {
+  public static Timestamp convertTimestampToZone(Timestamp ts, ZoneId fromZone, ZoneId toZone,
+      boolean legacyConversion) {
+    if (legacyConversion) {
+      try {
+        DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
should this be a threadlocal value? using this way is pretty slow iirc..

Reply:
Yes, that's right. I'll change it in follow-up commit.